app/vmalert/remotewrite/init.go

```diff
@@ -36,8 +37,8 @@ var (
 
 	maxQueueSize  = flag.Int("remoteWrite.maxQueueSize", 1e6, "Defines the max number of pending datapoints to remote write endpoint")
 	maxBatchSize  = flag.Int("remoteWrite.maxBatchSize", 1e4, "Defines max number of timeseries to be flushed at once")
-	concurrency   = flag.Int("remoteWrite.concurrency", 4, "Defines number of writers for concurrent writing into remote write endpoint")
-	flushInterval = flag.Duration("remoteWrite.flushInterval", 5*time.Second, "Defines interval of flushes to remote write endpoint")
+	concurrency   = flag.Int("remoteWrite.concurrency", cgroup.AvailableCPUs()*2, "Defines number of writers for concurrent writing into remote write endpoint")
```

Comment:
We already have `defaultConcurrency` defined but doesn't affect actual concurrency in default case.

https://github.com/VictoriaMetrics/VictoriaMetrics/blob/e347d905313cd1100b16749ed630d00387963615/app/vmalert/remotewrite/client.go#L26

How about changing the `defaultConcurrency` to `cgroup.AvailableCPUs()*2` and setting the default for the flag `-remoteWrite.concurrency` to 0?

Reply:
`cgroup.AvailableCPUs()*2` expression can't be set to const. 
Please note that default constants are used only in case of misconfiguration (and in tests).

Reply:
Yeah, we can change it to a variable. My point is `defaultConcurrency` is not applied in default case as its name suggests now, since flag `-remoteWrite.concurrency` already has a default value.
And if `-remoteWrite.concurrency` is misconfigured, setting `defaultConcurrency` to `cgroup.AvailableCPUs() * 2` aligns better with the intent.

Reply:
See updates in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/7272/commits/e32233b7eeee6d1d996794ce57ca0160b9493354

Reply:
Is it possible to correct the `vmalert_remotewrite_concurrency` when `-remoteWrite.concurrency` is misconfigured?
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/a06d52c4e81f7372337e4446a4373312308625e6/app/vmalert/remotewrite/client.go#L210-L212

Reply:
It is, but I don't think it is worth it. Moreover, this metric would help us to see misconfiguration earlier.
Maybe, we should drop `default_` constants completely, but this will be kinda not backward compatible.

Reply:
>Moreover, this metric would help us to see misconfiguration earlier.

In order to see misconfiguration earlier, we should break the backward compatiblity:)